### PR TITLE
Add scale::Output to Wrap (no_std) + minor improvements

### PR DESCRIPTION
**Pull request by Robbepop**
- Add `scale::Output` impl to `Wrap` in `no_std`
- Minor improvements

### CODE REVIEW

File: core/src/hash/accumulator.rs [hunk 1]
```diff
@@ -103,6 +120,7 @@ impl<'a> Accumulator for Wrap<'a> {
     }
 
     fn write(&mut self, bytes: &[u8]) {
+        assert!(self.len() + bytes.len() <= self.capacity());
```
Comment (athei) [inline core/src/hash/accumulator.rs:123]: Isn't that redundant? Your slicing operation will panic anyways when this happens.

File: core/src/hash/accumulator.rs [hunk 2]
```diff
@@ -113,3 +131,16 @@ impl<'a> Accumulator for Wrap<'a> {
...
 }
+
+#[cfg(not(feature = "std"))]
+impl<'a> scale::Output for Wrap<'a> {
+    fn write(&mut self, bytes: &[u8]) {
+        <Self as Accumulator>::write(self, bytes)
+    }
+
+    fn push_byte(&mut self, byte: u8) {
+        assert!(self.len() < self.capacity());
```
Comment (athei) [inline core/src/hash/accumulator.rs:142]: As above. I think this is just redundant noise (IMHO).

File: core/src/hash/accumulator.rs (hunk 1 above)
Comment (Robbepop) [reply core/src/hash/accumulator.rs:123]: true

File: core/src/hash/accumulator.rs (hunk 2 above)
Comment (Robbepop) [reply core/src/hash/accumulator.rs:142]: true

**Pull request merged**